app/scripts/metamask-controller.js

```diff
@@ -5501,13 +5526,12 @@ export default class MetamaskController extends EventEmitter {
       useRequestQueue: this.preferencesController.getUseRequestQueue.bind(
         this.preferencesController,
       ),
+      // TODO: Should this be made async in queued-request-controller package?
+      // Doing so allows us to DRY up getPermittedAcounts and getPermittedAccountsSorted
       shouldEnqueueRequest: (request) => {
         if (
           request.method === 'eth_requestAccounts' &&
-          this.permissionController.hasPermission(
-            request.origin,
-            PermissionNames.eth_accounts,
-          )
+          this.getPermittedAccounts().length > 0
         ) {
           return false;
         }
```

Comment:
thoughts?

Reply:
Why does it allow us to DRY them?

Reply:
the current `QueuedRequestMiddleware` implementation requires that `shouldEnqueueRequest` return synchronously. `getPermittedAcounts` is synchronous, but returns accounts potentially in the incorrect lastSelected order. `getPermittedAccountsSorted` is async, but returns accounts in the correct order. `getPermittedAcounts` only exists right now because it is used in `shouldEnqueueRequest`, otherwise `getPermittedAccountsSorted` could just replace `getPermittedAccounts`

Reply:
Alright. Lets maybe leave this as a future enhancement!